lib/wasix/src/runtime/package_loader/load_package_tree.rs

```diff
@@ -213,13 +217,32 @@ fn legacy_atom_hack(
         "(hack) The command metadata is malformed. Falling back to the first atom in the WEBC file",
     );
 
+    let hash = atom_hash(webc, &name);
+
     Some(BinaryPackageCommand::new(
         command_name.to_string(),
         metadata.clone(),
         atom,
+        hash,
     ))
 }
 
+fn atom_hash(webc: &Container, atom_name: &str) -> ModuleHash {
+    let base64_encoded = webc.manifest().atoms[atom_name]
+        .signature
+        .strip_prefix("sha256")
+        .expect("malformed atom signature: does not have sha256 prefix");
+    let sha256 = base64::prelude::BASE64_STANDARD
+        .decode(base64_encoded)
+        .expect("malformed base64 encoded hash");
+    let hash: [u8; 32] = sha256
+        .as_slice()
+        .try_into()
+        .expect("sha256 hash is not 32 bytes");
+
+    ModuleHash::sha256_from_bytes(hash)
```

Comment:
I think this is quite hacky. We should just have a `FromStr` / `IntoStr` impl for the `ModuleHash`, that simply can do the `sha256:abc` -> `ModuleHash::SHA256(..)` conversion and back.

Reply:
There are no hacks here. I'm just doing the steps that were taken in `pirita`, but in reverse. But I agree that your way may be more _convenient_.

Reply:
Let's do what's more convenient, is better and we can reuse it in more places. Maybe we can even add it to wasmer-config. In any case, let's do it proper

Reply:
This should not panick, but return an error instead...